Sources/Controllers/2.0/Settings/NavigationSettings/OAVehicleParametersSettingsViewController.mm

```diff
@@ -0,0 +1,371 @@
+        return cell;
+    }
+    else if ([cellType isEqualToString:@"OAHorizontalCollectionViewCell"])
+    {
+        static NSString* const identifierCell = @"OAHorizontalCollectionViewCell";
+        OAHorizontalCollectionViewCell* cell = [tableView dequeueReusableCellWithIdentifier:identifierCell];
+        if (cell == nil)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:identifierCell owner:self options:nil];
+            cell = (OAHorizontalCollectionViewCell *)[nib objectAtIndex:0];
+            cell.delegate = self;
+            cell.selectionStyle = UITableViewCellSelectionStyleNone;
+        }
+        if (cell)
+        {
+            cell.dataArray = item[@"values"];
+            cell.selectedIndex = [item[@"selectedValue"] intValue];
+            [cell.collectionView reloadData];
+            [cell layoutIfNeeded];
+        }
+        return cell;
+    }
+    return nil;
+}
+
+- (UIView *) tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section
+{
+    if (section == kDescriptionStringSection)
+    {
+        NSString *parameter = _vehicleParameter[@"name"];
+        NSString *descriptionString = [self getParameterDescription:parameter];
+        CGFloat heightForHeader = [self heightForLabel:descriptionString];
+        UIView *vw = [[UIView alloc] initWithFrame:CGRectMake(0, 0.0, tableView.bounds.size.width - OAUtilities.getLeftMargin * 2, heightForHeader)];
+        CGFloat textWidth = self.tableView.bounds.size.width - (kSidePadding + OAUtilities.getLeftMargin) * 2;
+        UILabel *description = [[UILabel alloc] initWithFrame:CGRectMake(kSidePadding + OAUtilities.getLeftMargin, 0.0, textWidth, heightForHeader)];
+        UIFont *labelFont = [UIFont systemFontOfSize:15.0];
+        description.font = labelFont;
+        [description setTextColor: UIColorFromRGB(color_text_footer)];
+        NSMutableParagraphStyle *style = [[NSMutableParagraphStyle alloc] init];
+        [style setLineSpacing:6];
+        description.attributedText = [[NSAttributedString alloc] initWithString:descriptionString attributes:@{NSParagraphStyleAttributeName : style}];
+        description.numberOfLines = 0;
+        [vw addSubview:description];
+        return vw;
+    }
+    return nil;
+}
+
+- (CGFloat) tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
+{
+    if (section == kDescriptionStringSection)
+    {
+        NSString *parameter = _vehicleParameter[@"name"];
+        NSString *descriptionString = [self getParameterDescription:parameter];
+        CGFloat heightForHeader = [self heightForLabel:descriptionString];
+        return heightForHeader + kSidePadding;
+    }
+    return 0.01;
+}
+
+- (NSInteger) tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return _data[section].count;
+}
+
+- (NSInteger) numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 2;
+}
+
+- (CGFloat) heightForLabel:(NSString *)text
```

Comment:
1. Similar method should be in OAUtilities.
2. Why UILabel is created at all? It is possible to calculate height with calculateTextBounds.